src/Workspaces/Core/Portable/Rename/RenameLocation.ReferenceProcessing.cs

```diff
@@ -448,93 +448,99 @@ internal static async Task<IEnumerable<RenameLocation>> GetRenamableReferenceLoc
+                ArrayBuilder<RenameLocation> stringAndCommentRenameLocationsBuilder,
+                CancellationToken cancellationToken)
             {
                 var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
                 var renameTextLength = renameText.Length;
 
-                var renameStringsAndPositions = root
-                    .DescendantTokens()
-                    .Where(t => syntaxFactsService.IsStringLiteralOrInterpolatedStringLiteral(t) && t.Span.Length >= renameTextLength)
-                    .Select(t => Tuple.Create(t.ToString(), t.Span.Start, t.Span));
-
-                if (renameStringsAndPositions.Any())
+                using var _1 = ArrayBuilder<(string rawText, TextSpan containingSpan)>.GetInstance(out var tokensOrTriviaToCheck);
+                // 1. Check all the tokens to rename in string, and the tokens in DocumentationCommentTrivia to rename in comments.
+                foreach (var token in root.DescendantTokens(descendIntoTrivia: renameInComments))
                 {
-                    AddLocationsToRenameInStringsAndComments(document, root.SyntaxTree, renameText,
-                        renameStringsAndPositions, renameLocations);
-                }
-            }
+                    if (token.Span.Length < renameTextLength)
+                    {
+                        continue;
+                    }
 
-            private static async Task AddLocationsToRenameInCommentsAsync(
-                Document document, string renameText, ArrayBuilder<RenameLocation> renameLocations, CancellationToken cancellationToken)
-            {
-                var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
-                var renameTextLength = renameText.Length;
+                    if (renameInStrings && syntaxFactsService.IsStringLiteralOrInterpolatedStringLiteral(token)
+                        || renameInComments && renameRewriterLanguageService.IsRenamableTokenInComment(token))
+                    {
+                        tokensOrTriviaToCheck.Add((rawText: token.ToString(), containingSpan: token.Span));
+                    }
+                }
 
-                var renameStringsAndPositions = root
-                    .DescendantTrivia(descendIntoTrivia: true)
-                    .Where(t => t.Span.Length >= renameTextLength)
-                    .Select(t => Tuple.Create(t.ToString(), t.Span.Start, t.Token.Span));
+                // 2. Check all the trivia if rename in comments.
+                if (renameInComments)
+                {
+                    foreach (var trivia in root.DescendantTrivia(descendIntoTrivia: true))
+                    {
+                        if (trivia.Span.Length >= renameTextLength)
+                        {
+                            tokensOrTriviaToCheck.Add((rawText: trivia.ToString(), containingSpan: trivia.Span));
```

Comment:
This is one of the core changes in this PR.
The trivia's rename location would be linked to its containing trivia, not its linked token.

Because in the old RenameRewriter, we rename the trivia when visiting the token.
If the trivia's location is not in the token's span (e.g. common single-line comment like //)
The regex is used twice in the string to compute the replacement text.

I change this so that the RenameRewriter would rewrite the comment when visit the trivia, not visit the token.
(I feel it's hard to expand the old logic here to rename multiple symbols)